compiler/hermes-std/src/ConsoleReporter.cpp

```diff
@@ -17,16 +17,68 @@
 #include "hermes/ConsoleReporter.h"
 
 #include <iostream>
+#include <cstdlib>
+#include <string>
 
 namespace hermes
 {
 
+static constexpr const char *kTermColorRedTextCode = "\033[0;31m";
+static constexpr const char *kTermColorGreenTextCode = "\033[0;32m";
+static constexpr const char *kTermColorOrangeTextCode = "\033[0;33m";
+static constexpr const char *kTermColorBlueTextCode = "\033[0;34m";
+static constexpr const char *kTermColorMagentaTextCode = "\033[0;35m";
+static constexpr const char *kTermColorCyanTextCode = "\033[0;36m";
+static constexpr const char *kTermColorWhiteTextCode = "\033[0;37m";
+
+static constexpr const char *kTermBoldTextCode = "\033[1m";
+static constexpr const char *kTermUnderlineTextCode = "\033[4m";
+static constexpr const char *kTermInverseTextCode = "\033[7m";
+static constexpr const char *kTermBoldOffTextCode = "\033[21m";
+static constexpr const char *kTermUnderlineOffTextCode = "\033[24m";
+static constexpr const char *kTermInverseOffTextCode = "\033[27m";
+
+static constexpr const char *kTermColorResetAllCode = "\033[0m";
+
 void ConsoleReporter::notify(const hermes::Message *m)
 {
+  const char *env_color_p = std::getenv("ONE_HERMES_COLOR");
+  if (env_color_p)
+  {
+    auto env_color_str = std::string(env_color_p);
+    if ((env_color_str == "1") or (env_color_str == "ON"))
+      _is_colored = true;
+  }
```

Comment:
(optional) (A) this routine is simple and I think it won't take that much time but will run in everly lines of message.
you can (1) add a TODO like "optimize this for speed" or (2) revise this code to check if not checked before
and (B) when turned on, there is no check to turn it off. I'm not sure when once turned one, we need to check for off but for safetly in usage case, we may need it. for this case, we may have to skip (A) case.

Anyway we can go as-is or just add some TODOs...
this library maybe used in our TVN backend and if something goes wrong, there will be some reports.

Reply:
I've tested if this code and found that million calls takes ~20 milliseconds on my hardware.
This should not be a big problem, at least for now =)